### PR TITLE
Add color themes functionality

### LINKED ISSUE #38: [Enhancement] Add site color themes (check meeting notes)
~~add a warning or default tab so people know the original idea~~

### PR DESCRIPTION
**Pull request by markbussard**
### Changes implemented:

1. Refactor and broke up main.scss into several files for better organization / readability and easier maintainability. Files broken up into (page specific styles) app.scss and index.scss, (styles for components used within both index.html and app.html) components.scss, and variables.scss for any hard coded styling values.
2. Added a new themes.scss file which contains colors for our themes via a map, a function for retrieving relevant theme colors, and a mixin which will provide the additional themed classes on compilation to css. Resource used for this functionality can be found [here](https://medium.com/@dmitriy.borodiy/easy-color-theming-with-scss-bc38fd5734d1)
3. Added new themes dropdown to settings modal and functionality for storing user themes in localStorage
4. Added initial dark theme

Closes #38

### CODE REVIEW

File: source/app.html [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
-    <link rel="stylesheet" href="./scss/main.scss">
-    <link rel="stylesheet" href="./fontawesome/css/all.min.css">
+    <link rel="stylesheet" href="scss/main.scss">
+    <link rel="stylesheet" href="fontawesome/css/all.min.css">
     <link rel="icon" href="img/green-tomato.ico">
     <link rel="manifest" href="manifest.json">
     <title>Electric Pomato</title>
+    <script>
```
Comment (alanlwang1) [inline source/app.html:13]: Should we add this to its on script file to avoid duplication?

Comment (markbussard) [reply source/app.html:13]: > Should we add this to its own script file to avoid duplication?

Yeah we probably can. I'll make the change and try it out

**Review by alanlwang1 — COMMENTED** [review]
For dark mode, Should we change the inputs to have a black background with white text?
image
Should we change the color of the `<h2>` elements and notification text to be white during dark mode as well?
image
We can maybe consider having a darker background for the modals that would make the white not stand out as much

**Comment by markbussard** [comment]
> For dark mode, Should we change the inputs to have a darker background with white text? image Should we change the color of the `<h2>` elements and notification text to be white during dark mode as well? image We can maybe consider having a darker background for the modals that would make the white not stand out as much

Went ahead and updated some of the theming for these, if you could take a look again when you get a chance.

**Review by HarrisSteven — APPROVED** [review]

**Pull request merged**